core/src/main/java/org/apache/calcite/prepare/CalciteCatalogReader.java

```diff
@@ -173,7 +174,14 @@ public CalciteCatalogReader withSchemaPath(List<String> schemaPath) {
   }
 
   public RelDataType getNamedType(SqlIdentifier typeName) {
-    return null;
+    Pair<CalciteSchema, String> pair =
+        Util.schema(getRootSchema(), ImmutableList.<String>of(), typeName);
+    CalciteSchema.TypeEntry typeEntry = pair.left.getType(pair.right, false);
+    if (typeEntry != null) {
```

Comment:
can we try to come up with some value instead of `null`

Reply:
I think that it's reasonable that `CalciteSchema.getType(String typeName, boolean caseSensitive)` returns null; it is consistent with `CalciteSchema.getTable`.